### DISCUSSION TITLE
Uncrustify or similar implementations of code style guidelines

### DISCUSSION THREAD

**Discussion by phretor** [jgromes/RadioLib#628] [Ideas]
Hi!

As one of the contributors of RadioLib, I'm often challenged by the code style guidelines (each one has their own preferences, of course).

I was thinking if it would make sense to implement the code style guidelines through something reproducible like Uncrustify. I'm not really a C++ dev, so I may be unaware of some de-facto standard tools in this space.

My 2c.

**Comment by jgromes** [comment]
To be honest, the main reason there is a code style section in the contributin doc at all is to attempt some consistency in the library code. I would not think of myself as a C++ developer, and since I'm mostly self-taught, my preferred coding style is just random bits and pieces stuck together that I picked up over the years.

Uncrustify looks good, I will try to find a bit of time to familiarize myself with it or explore other options, since whatever coding style might be preferred, it's definitely better to have it enforced by something that can be automated.

**Comment by phretor** [comment]
Sounds good! Happy to beta test it :-)

FYI, here's an example project that I know it uses Uncrustify: https://github.com/eProsima/cpp-style

**Reply by jgromes** [reply]
Had a bit of time, so there's now a test on the [uncrustify branch](https://github.com/jgromes/RadioLib/tree/uncrustify). I tried using an existing GitHub action (https://github.com/coleaeason/actions-uncrustify) but it seems to be broken as can be seen in the recent CI runs (https://github.com/jgromes/RadioLib/actions/runs/3981080312).

**Reply by phretor** [reply]
It sorts of run now, right?
